### DISCUSSION TITLE
`runhaskell` test failure PostgREST 11.2

### DISCUSSION THREAD

**Discussion by Vekhir** [PostgREST/postgrest#2950] [Q&A] [answered]
There is a testfailure with postgrest 11.2:

```
psql:test/spec/fixtures/roles.sql:6: ERROR:  syntax error at or near ":"
LINE 1: ...stgrest_test_default_role, postgrest_test_author TO :PGUSER;
```
I'm not sure what exactly the issue here is, but the tests shouldn't fail anyhow.

### Environment
OS: Arch Linux
Kernel: Linux 6.4.12-arch1-1
GHC: 9.0.2
postgrest: 11.2.0-1
postgresql: 15.4-2

### Description of issue

When running postgrest 11.2.0 with the following commandline:
```
    eval $(pifpaf run postgresql --host 127.0.0.1)
    psql -v ON_ERROR_STOP=1 -f test/spec/fixtures/load.sql
    runhaskell Setup test --show-details=direct
```

There is a test failure with the following error:
```
CREATE EXTENSION
ALTER DATABASE
SET
DROP SCHEMA
DROP TYPE
DROP ROLE
CREATE ROLE
CREATE ROLE
CREATE ROLE
psql:test/spec/fixtures/roles.sql:6: ERROR:  syntax error at or near ":"
LINE 1: ...stgrest_test_default_role, postgrest_test_author TO :PGUSER;
```

The entire file (`test/spec/fixtures/roles.sql`) looks like this:
```
DROP ROLE IF EXISTS postgrest_test_anonymous, postgrest_test_default_role, postgrest_test_author;
CREATE ROLE postgrest_test_anonymous;
CREATE ROLE postgrest_test_default_role;
CREATE ROLE postgrest_test_author;

GRANT postgrest_test_anonymous, postgrest_test_default_role, postgrest_test_author TO :PGUSER;
```

Do you have an idea what the reason is or how to fix it?

Thanks,
-- Vekhir

**Comment by steve-chavez** [comment, marked as answer]
Can you use Nix to setup a dev environment and run the tests? https://github.com/PostgREST/postgrest/blob/main/nix/README.md

That should ensure PGUSER is set. 

Otherwise try something like:

https://github.com/PostgREST/postgrest/blob/60fe59c3fb1e62eefc0262ed661acaf3ce381647/nix/tools/withTools.nix#L95

That is, set the missing variable with `-v PGUSER=".."`.
